cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -237,10 +277,24 @@ def from_unitary(u: np.ndarray) -> Optional['SingleQubitCliffordGate']:
         z_to = _to_pauli_transform(u @ z @ u.conj().T)
         if x_to is None or z_to is None:
             return None
-        return SingleQubitCliffordGate.from_double_map({pauli_gates.X: x_to, pauli_gates.Z: z_to})
+        return SingleQubitCliffordGate.from_cliiford_tableau(
+            _to_cliiford_tableau(x_to=x_to, z_to=z_to)
+        )
 
     def transform(self, pauli: Pauli) -> PauliTransform:
-        return self._rotation_map[pauli]
+        x_to = self._clifford_tableau.destabilizers()[0]
+        z_to = self._clifford_tableau.stabilizers()[0]
+        if pauli == pauli_gates.X:
+            to = x_to
+        elif pauli == pauli_gates.Z:
+            to = z_to
+        else:
```

Comment:
where is the fourth branch? Can't it be identity?

Reply:
In Single Qubit Clifford gate, it is not possible because it is automorphism (`I` always maps to `I`). So x_to and z_to should never map to `I`.
However, it is possible for multiple qubits case, say, we can map `XX` into  `XI`.